Marlin/printcounter.cpp

```diff
@@ -103,35 +125,59 @@ void PrintCounter::showStats() {
     SERIAL_ECHOPGM(")");
   #endif
 
-  // @todo longestPrint missing implementation
+  uint32_t l = this->data.longestPrint / 60;
+  SERIAL_ECHOPGM(", Longest job: ");
+
+  SERIAL_ECHO(l / 60 / 24);
+  SERIAL_ECHOPGM("d ");
+
+  SERIAL_ECHO((l / 60) % 24);
+  SERIAL_ECHOPGM("h ");
+
+  SERIAL_ECHO(l % 60);
+  SERIAL_ECHOPGM("min");
```

Comment:
`d h min` instead of `d h m`?

Lately I've been favoring `d:hh:mm:ss` for times, as they are straightforward and compact. Well, actually with `M31` I went with…
```cpp
inline void gcode_M31() {
  millis_t t = print_job_timer.duration();
  int d = int(t / 60 / 60 / 24),
      h = int(t / 60 / 60) % 60,
      m = int(t / 60) % 60,
      s = int(t % 60);
  char time[30];
  if (d)
    sprintf_P(time, PSTR("%i:%02i:%02i:%02i"), d, h, m, s);
  else if (h)
    sprintf_P(time, PSTR("%02i:%02i:%02i"), h, m, s);
  else
    sprintf_P(time, PSTR("%02i:%02i"), m, s);

  lcd_setstatus(time);

  SERIAL_ECHO_START;
  SERIAL_ECHOPGM(MSG_PRINT_TIME " ");
  SERIAL_ECHOLN(time);

  thermalManager.autotempShutdown();
}
```